db/indexes.go

```diff
@@ -194,7 +194,7 @@ func (i *SGIndex) createIfNeeded(bucket *base.CouchbaseBucketGoCB, useXattrs boo
 	if numReplica > 0 || (useXattrs && i.shouldIndexTombstones()) {
 		options = &base.N1qlIndexOptions{
 			NumReplica:      numReplica,
-			IndexTombstones: i.shouldIndexTombstones(),
+			IndexTombstones: useXattrs && i.shouldIndexTombstones(),
```

Comment:
Can you add a comment somewhere that explains why indexing tombstones are only needed if `useXattrs` is set to true?

Alternatively, what about changing it to:

```
IndexTombstones: i.shouldIndexTombstones(useXattrs)
```

and having `i.shouldIndexTombstones()` return false if `useXattrs != true`?   I think the advantage there is that any code that needs to make that decision will be forced to think about about whether xattrs are enabled or not.

Reply:
There's already a comment included w/ the associated flag:
	IdxFlagIndexTombstones                           // When xattrs=true, index should be created with {“retain_deleted_xattr”:true} in order to index tombstones

I'll add an additional comment w/ shouldIndexTombstones.